packages/template-retail-react-app/app/components/display-price/index.jsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React from 'react'
+import PropTypes from 'prop-types'
+import {Skeleton, Text} from '@salesforce/retail-react-app/app/components/shared/ui'
+import {useIntl} from 'react-intl'
+import {useCurrency} from '@salesforce/retail-react-app/app/hooks'
+
+const DisplayPrice = ({
+    basePrice,
+    discountPrice,
+    isProductASet = false,
+    currency,
+    discountPriceProps,
+    basePriceProps,
+    skeletonProps
+}) => {
+    const intl = useIntl()
+    const {currency: activeCurrency} = useCurrency()
+    return (
+        <Skeleton isLoaded={basePrice} display={'flex'} {...skeletonProps}>
+            <Text fontWeight="bold" fontSize="md" mr={1}>
+                {isProductASet &&
+                    `${intl.formatMessage({
+                        id: 'product_view.label.starting_at_price',
+                        defaultMessage: 'Starting at'
+                    })} `}
+            </Text>
+            {discountPrice > 0 && (
```

Comment:
```suggestion
            {typeof discountPrice === 'number' && (
```
$0 (free) is a valid discount price! Could also extract this to a `hasDiscountPrice` variable.

Reply:
If discountPrice is 0 in the promotion, it will be filter out [here](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/1455/files#diff-cbbaa6eb4930763d665494ecf261cfda4a16130a360ff028ee11560d8b3490ebR46), and returns as null. I think we are safe

Reply:
Oh, then I guess also update the filter to check that the price is a number! We don't want to prevent $0 as a discount price.

Reply:
although 0 is a valid discount price, but it does not make sense on the business POV. If discount is 0, do we need to display it?

Reply:
If a business wants to offer a product for free (or even a negative price!), we should allow them to make that decision. How they price things is their responsibility, not ours.

Reply:
Good point. If that is the case, I've made change to to getDisplayPrice to only filter out null and undefined values, we should keep the 0 if applicable